extensions/pkg/terraformer/terraformer.go

```diff
@@ -178,50 +183,56 @@ func (t *terraformer) execute(ctx context.Context, command string) error {
 	// because of syntax errors. In this case, we want to skip the Terraform destroy pod (as it wouldn't do anything
 	// anyway) and just delete the related ConfigMaps/Secrets.
 	if command == "destroy" {
-		skipApplyOrDestroyPod = t.IsStateEmpty()
+		skipApplyOrDestroyPod = t.IsStateEmpty(ctx)
 	}
 
 	if !skipApplyOrDestroyPod {
 		// Create Terraform Pod which executes the provided command
 		generateName := t.computePodGenerateName(command)
+
+		logger.Info("Deploying Terraformer Pod", "generateName", generateName)
 		pod, err := t.deployTerraformerPod(ctx, generateName, command)
 		if err != nil {
-			return fmt.Errorf("failed to deploy the Terraformer Pod with .meta.generateName '%s': %s", generateName, err.Error())
+			return fmt.Errorf("failed to deploy the Terraformer Pod with .meta.generateName %q: %w", generateName, err)
```

Comment:
Why are you switching to `%w`? It is not looking very good to me.

```golang
package main

import (
	"fmt"
)

func main() {
	const name, id = "bueller", 17
	err := fmt.Errorf("user %q (id %d) not found", name, id)
	fmt.Printf("%w\n", err) // %!w(*errors.errorString=&{user "bueller" (id 17) not found})
	fmt.Printf("%v\n", err) // user "bueller" (id 17) not found
}
```
The above code snippet is taken from [here](https://golang.org/pkg/fmt/#Errorf)

Reply:
`%w` is only working with `fmt.Errorf`: https://golang.org/pkg/fmt/#Errorf
That's why it's looking strange with `fmt.Printf` (behaves like `%+v`, I believe).

E.g.:
```go
const name, id = "bueller", 17
err := fmt.Errorf("user %q (id %d) not found", name, id)
fmt.Println(fmt.Errorf("wrapped error: %w", err)) // wrapped error: user "bueller" (id 17) not found
```

Reply:
Well the printing should be the same. Snippet for reference:

```go
package main

import (
	"fmt"
)

func main() {
	const name, id = "bueller", 17
	err := fmt.Errorf("user %q (id %d) not found", name, id)
	errOut1:= fmt.Errorf("%v",err)
	errOut2:= fmt.Errorf("%w",err)
	fmt.Printf("%v\n", errOut1) //prints user "bueller" (id 17) not found
	fmt.Printf("%v\n", errOut2) //user "bueller" (id 17) not found
}
```

@timebertt You should use `%w` only if you  want the callers to have access to the original error. If you think that the callers cannot use the internal error then wrap it with `%v`

```go
return fmt.Errorf("failed to deploy the Terraformer Pod with .meta.generateName %q: %v", generateName, err)
```

Reply:
Thanks for the explanations :heart: .

Reply:
> @timebertt You should use `%w` only if you want the callers to have access to the original error. If you think that the callers cannot use the internal error then wrap it with `%v`

I guess, in general it makes sense to wrap errors, that will be propagated up to other components (as it's the case here).
They might decide to actually unwrap the errors at some point.